examples/mac-example.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2020 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#include <mococrw/mac.h>
+#include <mococrw/util.h>
+#include <iostream>
+
+using namespace mococrw;
+
+std::unique_ptr<mococrw::HMAC> doHmac(DigestTypes digestType, const std::vector<uint8_t> &key,
+                                      const std::vector<uint8_t> &message)
+{
+    auto hmac = std::make_unique<mococrw::HMAC>(digestType, key);
+    hmac->update(message);
+    return hmac;
+}
+
+void verify(DigestTypes digestType, const std::vector<uint8_t> &key, const std::vector<uint8_t> &message,
+            const std::vector<uint8_t> &authenticationTag)
+{
+    auto hmac = doHmac(digestType, key, message);
+    auto tag = hmac->finish();
+
+    try {
+        hmac->verify(authenticationTag);
+    } catch (const MoCOCrWException &e) {
+        std::cerr << "Verification failed." << std::endl;
+        std::cerr << e.what();
+        exit(EXIT_FAILURE);
+    }
+
+    std::cout << "Verification successful." << std::endl;
+}
+
+std::vector<uint8_t> calculate(DigestTypes digestType, const std::vector<uint8_t> &key,
+                               const std::vector<uint8_t> &message)
+{
+    auto hmac = doHmac(digestType, key, message);
+    return hmac->finish();
+}
+
+int main(void)
+{
+    DigestTypes digestType = DigestTypes::SHA512;
+    std::vector<uint8_t> message = utility::fromHex("deadbeef");
+    std::vector<uint8_t> key = utility::fromHex("beefdead");
+
+    auto tag = calculate(digestType, key, message);
```

Comment:
Just demo this here all at once. It is in principle 3 lines. Now I need to read 2 addional functions to understand 3 LOC